### PR TITLE
Update to RxJS 5

### LINKED ISSUE #171: RxJS 5 support for rx-recompose
Is `rx-recompose` going to get support for RxJS 5 anytime soon? I would like to use `rx-recompose` but am unsure if it interops cleanly.

### PR DESCRIPTION
**Pull request by calebmer**
Closes #171.

### CODE REVIEW

File: src/packages/rx-recompose/createComponent.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Component } from 'react'
-import { Observable } from 'rx'
+import { Observable } from 'rxjs'
```
Comment (wuct) [inline src/packages/rx-recompose/createComponent.js]: We can import only Observable

```JavaScript
import { Observable } from 'rxjs/Observable'
```

for bundle size optimization

**Comment by acdlite** [comment]
Thanks for the PR but I want to wait until RxJS 5 is out of beta. At that time, we'll drop the dependency on RxJS and make it work with any spec-compliant observable. I'm also considering retiring rx-recompose entirely and bringing it into the main project.

See the rx-recompose-next branch for peek at how this will look. A bit out of date, but some version of that is what will eventually be released.

**Comment by calebmer** [comment]
That's a good goal. While RxJS isn't out of beta, the spec *is* finalized. Whether this means creating a new `recompose-observable` or rolling it into `recompose` can be done now for the people using the spec. RxJS 5 is in a pretty stable condition at the moment.

I'd like to see the plan to update this module implemented soon for early adopters. What can I do to help?

**Comment by istarkov** [comment]
I also see that RxJs 5 is more widely used in that projects I'm interested in, so also think that even it is in beta it will be the best choice to use it now.

**Comment by chicoxyzzy** [comment]
fyi while rxjs 5 api is frozen the library is not stable enough yet

**Comment by acdlite** [comment]
@calebmer I went ahead and merged your branch into `rx-recompose-next`. Added an additional commit to remove the dependency on RxJS: https://github.com/acdlite/recompose/commit/5214295d12f1c3ef23e98a734a51cdba6e5ffa2e

Care to review it? Then I'll publish it on npm with a `next` tag.

@chicoxyzzy Thanks for volunteering! Care to review as well?

**Comment by chicoxyzzy** [comment]
yep. I'll try this branch but not earlier than week after ReactEurope

**Pull request closed without merging**